stl/inc/algorithm

```diff
@@ -3321,6 +3358,26 @@ _NODISCARD _CONSTEXPR20 _FwdIt1 find_first_of(
     const auto _ULast1  = _STD _Get_unwrapped(_Last1);
     const auto _UFirst2 = _STD _Get_unwrapped(_First2);
     const auto _ULast2  = _STD _Get_unwrapped(_Last2);
+#if _USE_STD_VECTOR_ALGORITHMS
+    constexpr ptrdiff_t _Threshold = 16; // vectorization is likely to be a win after this size

```

Comment:
Question: Is this intentionally an element threshold (versus bytes)?

Reply:
I observe that for vectored implementation the constant cost part is big, and per loop iteration cost is small. Which is kind of expected. We compare that with scalar where per-element cost is big. So it is elements.

Reply:
This vector algorithm has unusually high constant cost. I suspect, based on what I observe in profiler, that `memcpy` is expensive here, which makes sense, because it should either branch on sizes, or have per-byre loop. Since we don't do reading beyond the range tricks (anymore), I don't see any faster `memcpy` alternatives.

The loop iteration is only moderately expensive, it has one unusually slow instruction (which is still very fast given how many comparisons it replaces), but the loop path consists of only that instruction (repeated intrinsic with other suffix letter queries different output of the already executed instruction) and pointer increment/end checking.

Reply:
Thanks, good to know! :bulb: